backend/src/main/scala/cromwell/backend/io/GlobFunctions.scala

```diff
@@ -0,0 +1,60 @@
+package cromwell.backend.io
+
+import java.nio.file.Files
+
+import cromwell.backend.BackendJobDescriptor
+import cromwell.core.CallContext
+import wdl4s.TaskCall
+import wdl4s.expression.{NoFunctions, PureStandardLibraryFunctionsLike}
+import wdl4s.values._
+import scala.collection.JavaConverters._
+import cromwell.core.path.PathImplicits._
+
+trait GlobFunctions extends PureStandardLibraryFunctionsLike {
+
+  def callContext: CallContext
+
+  def findGlobOutputs(call: TaskCall, jobDescriptor: BackendJobDescriptor): Set[WdlGlobFile] = {
+    val globOutputs = call.task.findOutputFiles(jobDescriptor.fullyQualifiedInputs, NoFunctions) collect {
+      case glob: WdlGlobFile => glob
+    }
+
+    globOutputs.distinct.toSet
+  }
+
+  def globDirectory(glob: String): String = globName(glob) + "/"
+  def globName(glob: String) = s"glob-${glob.md5Sum}"
+
+  /**
+    * Returns a path to the glob using toRealString.
+    *
+    * NOTE: Due to use of toRealString, returned paths must be passed to PathBuilders.buildPath, and will not work with
+    * Paths.get.
+    *
+    * This path is usually passed back into the glob() method below.
```

Comment:
These two provisos make me feel like adding a @geoffjentry -like comment, along the lines of "well if it needs special handling it should be wrapped in an ADT"